### DISCUSSION TITLE
Avalonia-11: Chrome-less window drop shadows

### DISCUSSION THREAD

**Discussion by ArchLeaders** [AvaloniaUI/Avalonia#9055] [General]
I recently moved my application to Avalonia-11, and I noticed that the drop-shadow on chrome-less (`ExtendClientAreaToDecorationsHint="True"`) windows had been removed (likely to allow for fully custom windows).

<img width="422" alt="image" src="https://user-images.githubusercontent.com/80713508/193116584-c7be1436-5939-41b4-8560-d5e8ff08b7cc.png">

Is there a property to restore the behavior from Avalonia-10 in Avalonia-11? (`ShowChromeBorder="True"`?)

**Comment by ArchLeaders** [comment]
Was doing some testing, and I randomly found out that if you use `dotnet publish` to build the executable, the drop shadow appears, but in any other case (even building in Release mode, running the exe from File Explorer, etc) it doesn't appear.

![image](https://user-images.githubusercontent.com/80713508/193156711-20a47bdb-2273-42b4-a698-05332fe7d04a.png)

The lower window is the normal code, but published, the top one is that same code, but when run normally from the `Release` folder, rather than `Release\net6.0\win-x64\publish`. If I had to guess, I'd imagine this is something to do with how cross-platform support is setup.

**Comment by ArchLeaders** [comment]
Ah okay, so you can get it to work in Debug/Release by just setting the runtime identifier. `<RuntimeIdentifier>win-x64</RuntimeIdentifier>`

**Comment by chenjing1294** [comment]
If you set `<RuntimeIdentifier>win10-x64</RuntimeIdentifier>`, the shadow will disappear. If you want to use WindowTransparencyLevel.AcrylicBlur or WindowTransparencyLevel.Mica must be set win10-x64. 😪

**Comment by vdm97** [comment]
I had the same problem when updating from Avalonia 0.10.19 to 11.0.0-preview7. I compile on Windows 10 with Visual Studio 2022 and my settings are `<TargetFramework>net7.0</TargetFramework>` and  `<RuntimeIdentifiers>win-x64;osx-x64;linux-x64</RuntimeIdentifiers>`. Changing `<RuntimeIdentifiers>` to `win-x64` or `win10-x64` did not help.  Trying different `TransparencyLevelHint` did not help either. Setting `ExtendClientAreaToDecorationsHint` to `false` did help but defeated the purpose... Lastly i changed `ExtendClientAreaChromeHints` from `PreferSystemChrome` to `SystemChrome` which fixed the issue but i do not know what other side effects this change has..

**Reply by vdm97** [reply]
Well, i mentioned the side effects.. Changing `ExtendClientAreaChromeHints` from `PreferSystemChrome` to `SystemChrome` makes the system buttons (close, minimize and maximize) transparent.. they are present and i can click on them but the icons are not present...

**Comment by YaMyha** [comment]
Hello! I had a problem: shadow from window bothered me to do rounded corners:
![изображение](https://github.com/AvaloniaUI/Avalonia/assets/115614725/cbf204cb-e64d-4140-8b23-f28f8bceac1b)
![изображение](https://github.com/AvaloniaUI/Avalonia/assets/115614725/85d571ad-91a0-41d0-bdf0-4215f4198361)
So I turned off  `HasSystemDecorations="False"` and got wanted result:
![изображение](https://github.com/AvaloniaUI/Avalonia/assets/115614725/d7e02bb2-4214-470f-ad7d-9950dc1dc727)
![изображение](https://github.com/AvaloniaUI/Avalonia/assets/115614725/78ef76aa-4e82-4860-ac08-e1e3d97d9f55)

Maybe it will help somebody

**Comment by Ryan-K-94** [comment]
Has anyone found a solution to this? Having the same issue since upgrading to V11.0.

**Reply by ArchLeaders** [reply]
The shadow border is OS specific, so if you build with the `win-x64` runtime identifier it should be restored. See https://github.com/AvaloniaUI/Avalonia/discussions/9055#discussioncomment-3767013 for details.

However, I did have some other issues as well, outlined in https://github.com/AvaloniaUI/Avalonia/issues/11685

**Reply by Ryan-K-94** [reply]
I tried putting `<RuntimeIdentifier>win-x64</RuntimeIdentifier>` in the property group section of the csproj file but it didn't fix it. Is that the right location for it?

**Reply by ArchLeaders** [reply]
That's correct. But also note that it does not work when `ExtendClientAreaToDecorationsHint` is `True`. The only way to have the shadow and remove the chrome is to set `SystemDecorations` to `BorderOnly`.

That has a host of other issues though, most of which are outlined in #11685. Also in versions `>11.0.1` this mode shows a weird partial white chrome with no buttons, and no size is applied to the window.

**Comment by Kelanor** [comment]
I think a solution is to move setting `SystemDecorations`  to `BorderOnly` into `Loaded` event implementation. I don't why but in constructor (or in .axaml file) it doesn't work.
I played with properties in DevTools and found that when I tick `ExtendClientAreaToDecorationsHint` (changing from `false` to `true` and vice versa) main window became shrinking and expanding. Only there I could only see border swadows. I think that this issue somehow involves rendering lifecycle...

**Comment by snowflysky** [comment]
I think this is a bug rather than a feature, and I have filed an issue #19679. In the meantime, I found a temporary solution in the sourceGit repository:[Fix drop shadow issue on Windows 10](https://github.com/sourcegit-scm/sourcegit/blob/fcc8928ac6cf0e47cce87177f092143a494bb44d/src/Native/Windows.cs#L59).
Note that the current code uses incorrect parameters when checking for Windows versions below Windows 10. I have submitted a PR: https://github.com/sourcegit-scm/sourcegit/pull/1818

**Comment by xmaxrayx** [comment]
no windows have 2 api for backdrop and other stuff then they said they gona make "official api" 
and windows 11 should be the target since most their API work focus on win11, 

more blame on Micrsoft than Avalonia for being spagatie desighn.

now for funny part i did this
```
        SystemDecorations="None"
        ExtendClientAreaToDecorationsHint="False"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="0"
        CanResize="False"
```
then applied corner DWM api and guess what they add shadow with even i didn't ask for it.

maybe avalnia team can give as ability to config window style and return hwnd to avalonia before init the window but  tbh microsoft being spagati override default value for no reason.

anyway you should think windows as is just a system with lazy replacemnet here and there and pray your thing look the same lmao.
